connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/PostgresIntegrationSuite.scala

```diff
@@ -123,4 +132,81 @@ class PostgresIntegrationSuite extends DockerJDBCIntegrationV2Suite with V2JDBCT
       )
     }
   }
+
+  test("SPARK-49162: Push down aggregate date_trunc function") {
+    def testAggregatePushdown(format: String, expectedResult: Set[Row]): Unit = {
+      val df = sql(
+        s"""
+            SELECT DATE_TRUNC('$format', time), COUNT(*)
```

Comment:
Is `DATE_TRUNC` supported by H2 database ?

Reply:
https://www.h2database.com/html/functions.html#date_trunc

Looks like yes, however the original task was adding support only for Postgres for now.

Reply:
I think we should support for H2 dialect first, then other database dialect.

Reply:
Added pushdown for H2 as well